[HV] 9.0.0.Beta3 (#203)
hibernate/in.relation.to: posts/MarkoB/2024-09-04-hibernate-validator-9-0-0-Beta3.adoc

= Hibernate Validator 9.0.0.Beta3 is out
Marko Bekhta
:awestruct-tags: [ "Hibernate Validator", "Releases" ]
:awestruct-layout: blog-post
:hibernate-validator-doc-url-prefix: https://docs.jboss.org/hibernate/validator/9.0/reference/en-US/html_single/
:hibernate-validator-jira-url-prefix: https://hibernate.atlassian.net/browse
:hibernate-validator-version-family: 9.0
:hibernate-validator-jira-project-id: 10060
:hibernate-validator-jira-version-id: 32328
---

We just published Hibernate Validator 9.0.0.Beta3,
the next beta release of the new 9.0 series of Hibernate Validator.

This series targets Jakarta EE 11. It is the implementation of the Jakarta Validation 3.1.

> Compared to the previous release in this series this version
> removes a few more constraints, configuration properties and APIs
> that have been deprecated for a few major versions now.
> We have also identified a few possible areas to improve while testing
> the previous version in the downstream projects.
> Also, there are a few fixes.

Reviewer: You used "a few" four times in three sentences :)

```suggestion
Compared to the previous release in this series, this version
removes a few more constraints, configuration properties and APIs
which have been deprecated for several major versions.
We have also identified some possible areas to improve while testing
the previous version in downstream projects.
Also, there are some bugfixes.
```

Author: just ... a few times 🙈🙈🙈🙈

Thanks for the suggestion!!! 😃

== What's new

[NOTE]
====
For a summary of all new features and improvements since 8.0,
head to https://hibernate.org/validator/releases/9.0/#whats-new[the dedicated page on hibernate.org].
====

=== Jakarta EE 11 and Jakarta Validation 3.1

> Just as a reminder: the 9.0 series is specifically intended to target the Jakarta EE 11 and implement the Jakarta Validation 3.1.

Reviewer: Lot of words here which aren't doing much useful work.

```suggestion
As a reminder: the 9.0 series targets Jakarta EE 11 and implements Jakarta Validation 3.1.
```

The 3.1 update of the specification does not bring many https://jakarta.ee/specifications/bean-validation/3.1/jakarta-validation-spec-3.1#whatsnew-31[changes]:

- The minimum required Java version is set to 17.

> - The rename of the specification from Jakarta Bean Validation to Jakarta Validation.
> - The clarification on the `record` validation, which has been supported by Hibernate Validator for some time.

Reviewer: ```suggestion
- The specification was renamed from _Jakarta Bean Validation_ to _Jakarta Validation_.
- Validation of Java `record` types was clarified (this has been supported by Hibernate Validator for some time).
```
